cmake/modules/AddPureSwift.cmake

```diff
@@ -65,6 +65,12 @@ function(_add_host_swift_compile_options name)
   target_compile_options(${name} PRIVATE
     $<$<COMPILE_LANGUAGE:Swift>:-color-diagnostics>
   )
+
+  if(LLVM_ENABLE_ASSERTIONS)
+    target_compile_options(${name} PRIVATE "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xcc -UNDEBUG>")
+  else()
+    target_compile_options(${name} PRIVATE "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xcc -DNDEBUG>")
+  endif()
```

Comment:
This is just a drive-by future proof fix. This ensures using the same `NDEBUG` between C/C++ and Swift.